facade/nodeFacade.go

```diff
@@ -391,49 +398,144 @@ func (nf *nodeFacade) GetNumCheckpointsFromAccountState() uint32 {
 }
 
 // GetProof returns the Merkle proof for the given address and root hash
-func (nf *nodeFacade) GetProof(rootHash string, address string) ([][]byte, error) {
-	rootHashBytes, err := hex.DecodeString(rootHash)
+func (nf *nodeFacade) GetProof(rootHash string, address string) (*shared.GetProofResponse, error) {
+	proof, value, err := nf.getProof(rootHash, address)
```

Comment:
previously, `nodeFacade` was only a middleware between the API and `node.go`. you introduced some logic here and also a marshalizer, that already exists in `node.go`. 
Maybe move the logic there?

Reply:
Done.